fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java

```diff
@@ -254,14 +267,38 @@ public static QueryBuilder toEsDsl(Expr expr, List<Expr> notPushDownList, Map<St
             }
         }
         if (expr instanceof InPredicate) {
-            return parseInPredicate(expr, column);
+            return parseInPredicate(expr, column, builderOptions);
         }
         if (expr instanceof FunctionCallExpr) {
             return parseFunctionCallExpr(expr);
         }
         return null;
     }
 
+    private static final DateTimeFormatter dorisFmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+    private static final DateTimeFormatter esFmt = ISODateTimeFormat.dateTime();
+
+    private static void setNeedCompatDate(String column, BuilderOptions builderOptions) {
+        Map<String, ObjectNode> originFields = builderOptions.getOriginFields();
```

Comment:
`OriginFields` is only used for set the `NeedCompatDate` filed, can we remote it? because it's not clear about the usage about these fileds.